[react-big-calendar] Fix/rbc outdated types

decimoseptimo opened the pull request:
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: in commits' messages

DangerBotOSS commented:
<!--
  0 failure: 
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-Danger;
-->



Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-big-calendar ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-big-calendar@latest/))
was missing the following properties:
1. components
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 08d27d26c26e5f179e6003851a0e49614747d6af
</p>

decimoseptimo commented:
ready for review

michalak111 commented on types/react-big-calendar/index.d.ts:
```diff
@@ -184,11 +184,18 @@ export interface ResourceHeaderProps {
+export interface DateCellWrapperProps {
+    range: Date[];
+    value: Date;
+    index: number;
```
please double confirm `index` is really prop of `dateCellWrapper` I just had quick look in implementation and couldn't find it

https://github.com/jquense/react-big-calendar/blob/ad8defa643692911fd0b00c71b70de94715140a9/src/BackgroundCells.js#L53

decimoseptimo commented:
Doesn't that very link proof it? wrapper has 3 props plus children.

On Mon, Jan 31, 2022, 11:42 PM TypeScript Bot ***@***.***>
wrote:

> @decimoseptimo <https://github.com/decimoseptimo> One or more reviewers
> has requested changes. Please address their comments. I'll be back once
> they sign off or you've pushed new commits. Thank you!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58380#issuecomment-1026553523>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEAXN4YVVST3AXDP4KOUNMTUY6FE5ANCNFSM5MSV65KQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

michalak111 commented:
```jsx
<Wrapper key={index} value={date} range={range}>
```

take a look, wrapper has two props `value` and `range`, `index` is used as a `key` which you don't have to put in props interface

decimoseptimo commented:
I see. I added it last minute, when I added children. Originally children
was throwing despite DateCellWrapper extending from react.componentType,
and i did not wanted the same issue to arise for index. But if it's not
necessary we can remove it. I'm not on a PC now though.

On Mon, Jan 31, 2022, 11:58 PM michalak111 ***@***.***> wrote:

> <Wrapper key={index} value={date} range={range}>
>
> take a look, wrapper has two props value and range, index is used as a key
> which you don't have to put in props interface
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58380#issuecomment-1026564329>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEAXN4ZQ4FVNYBQECFNJWQLUY6HCHANCNFSM5MSV65KQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

jakebailey commented:
This PR seems like it contains mostly ineffectual style changes (reflowing lines, replacement of `,` with `;`).

I'm not sure why all of the `| undefined` are going away; they were explicitly added in #54349 for TS 4.4. Do you have some case where this broke something or shows that they shouldn't be there?

decimoseptimo commented:
Most commits are for updating types, that's the reasoning for this PR. Although most code  lines come from [styling ](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58380/commits/394e4ba7fc64c53ff9b73c454e57e31bab7c607f) and [handling optional parameters](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58380/commits/08d27d26c26e5f179e6003851a0e49614747d6af).  The reasoning for the later is in its commit message (brakes TSLint).

jakebailey commented:
That reasoning you linked is valid for function arguments (if they have `?` then they don't need `| undefined`), whereas the bulk of the changes in that commit are to object properties where `?` doesn't mean the same thing.

decimoseptimo commented:
Well, the [CI was failing](https://github.com/DefinitelyTyped/DefinitelyTyped/commit/92eb96f75d615878fe172490d719f576fc36ecc3/checks?check_suite_id=5029579980) up until that commit.

jakebailey commented:
Yes, but that was because of the code on line 368 at the time of the CI failure (see the error message pointing to just that line), which was:

```ts
    onRangeChange?: (range: Date[] | { start: Date; end: Date }, view?: View | undefined) => void;
```

and should have been:

```ts
    onRangeChange?: (range: Date[] | { start: Date; end: Date }, view?: View) => void;
```

I would suggest reverting all of the `| undefined` changes to keep the same pattern as the existing code, and just ensure that particular `| undefined` remains fixed.

jakebailey reviewed (changes requested):
Marking as needing changes pending the above. I would restore `| undefined` back where it was removed, and any additional non-any properties added to objects should generally match the same pattern as the existing properties.

jakebailey commented on types/react-big-calendar/index.d.ts:
```diff
@@ -284,31 +305,50 @@ export interface Messages {
+    };
 }
 
-export type Culture = string;
```
I suspect that this is a breaking change because it is exported; what is the rationale for eliminating this type?

decimoseptimo replied:
I would rather question what is the rationale for it to exist

jakebailey replied:
I don't have context for why it was written this way, but my role in reviewing this is to try and prevent unnecessary breaks for people who are using the types now, and deleting an exported type is one that I suspect will cause issues for people who have imported it.

decimoseptimo replied:
I can see how removing an exported member is BC. I removed it because i dont see the value of it while it's just pointing to a primitive. Also there are mixed assignaments of culture, some pointiring to `Culture` type while others to [string | undefined](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-big-calendar/index.d.ts#L358). Can we just bump it?

jakebailey replied:
You're right that it is inconsistent. I am curious what the owners think, though I think only @michalak111 gave a review.

michalak111 replied:
I'd prefer to leave the `Culture` type and in places [string | undefined](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-big-calendar/index.d.ts#L358) update them to `Culture | undefined`

jakebailey reviewed (commented):
Thanks for reverting that change, much easier to see what changed.

jakebailey reviewed (commented):
I can't comment on any of the changes to do with `stringOrDate` to `Date`, but you may need to add `| undefined` to all of the optional properties you are adding.

decimoseptimo commented on types/react-big-calendar/index.d.ts:
```diff
@@ -242,12 +263,12 @@ export interface EventProps<TEvent extends object = Event> {
     isRtl: boolean;
     getters: {
-        eventProp?: EventPropGetter<TEvent> | undefined;
+        eventProp?: EventPropGetter<TEvent>;
```
solved

andrewbranch reviewed (changes requested):
I think you may need to update the version string in the header comment, unless you’re sure all these changes apply to v0.36. Current is v0.40.

decimoseptimo commented:
@andrewbranch wachu mean by 'current', master is 0.36, where i branched out from.

andrewbranch commented:
<img width="404" alt="Screen Shot 2022-03-28 at 4 55 53 PM" src="https://user-images.githubusercontent.com/3277153/160506209-168e5b55-0a2e-4ebc-913f-6e408d3db193.png">

decimoseptimo commented:
I see. There have been several RBC releases since 0.36, as far as DT goes, i'm not introducing any new changes. So i don't see the need to bump up.

andrewbranch commented:
So everything you’re changing was simply wrong at the time it was written for v0.36? Because the title says “outdated types,” not “incorrect types.”

decimoseptimo commented:
The latest RBC i tested on was 0.38.5, Bump to that sounds good?

andrewbranch commented:
Last question @decimoseptimo, since none of the owners were willing to review this PR, do you want to add yourself as an owner to help out with future reviews? You seem to have done a good bit of research in order to make these changes and your input would be valuable on future changes.

andrewbranch reviewed (approved):
Thanks!

decimoseptimo commented:
Ready to merge
